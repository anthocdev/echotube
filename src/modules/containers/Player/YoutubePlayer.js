import React from "react";
import Youtube from "react-youtube";
import { connect } from "react-redux";
import { setPlayingIndex } from "../../../actions/playback";

/* YouTube video player instance*/
class YoutubePlayer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      index: 0
    };
  }

  componentDidMount() {}

  render() {
    //Defining options for YouTube player
    const opts = {
      height: 320,
      width: 540,
      playerVars: {
        autoplay: 1
      }
    };
    return (
      <div>
        <Youtube
          videoId={this.props.videoId} //Video ID
          opts={opts} //Options
          onReady={this.onReady} //When player is ready
          onEnd={this.onEnd} //When video ends
        />
      </div>
    );
  }

  onReady(event) {
    event.target.playVideo();
  }

  onEnd(event) {
    //To be implemented -> Play next video
  }
}

function mapStateToProps(state, props) {
  return {
    playbackItems: state.player.playbackItems,
    currentIndex: state.player.currentIndex
  };
}

function mapDispatchToProps(dispatch) {
  return {
    dispatch: function(task) {
      dispatch(task);
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(YoutubePlayer);
