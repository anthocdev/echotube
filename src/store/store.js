import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
