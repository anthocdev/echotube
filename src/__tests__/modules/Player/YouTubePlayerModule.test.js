import YoutubePlayer from "../../../modules/containers/Player/YoutubePlayer";

import React from "react";
import { Provider } from "react-redux";
import store from "../../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("YouTube Player Module Testing", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <Provider store={store}>
          <YoutubePlayer />
        </Provider>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <YoutubePlayer />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
    console.log(root.toJSON());
  });
});
