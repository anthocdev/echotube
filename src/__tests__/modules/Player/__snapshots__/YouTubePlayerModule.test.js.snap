// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTube Player Module Testing Renders Matches Snapshot 1`] = `
<div>
  <div
    className=""
  >
    <div
      className={null}
      id={null}
    />
  </div>
</div>
`;
